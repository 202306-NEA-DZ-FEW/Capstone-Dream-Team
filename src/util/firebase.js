// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getFirestore } from "firebase/firestore";
import { getAuth } from "firebase/auth";
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
const firebaseConfig = {
    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
    authDomain: "capstone-project-944c9.firebaseapp.com",
    projectId: "capstone-project-944c9",
    storageBucket: "capstone-project-944c9.appspot.com",
    messagingSenderId: "455661891383",
    appId: "1:455661891383:web:65bbc812478ffda076b4b2",
};

// Initialize Firebase
export const app = initializeApp(firebaseConfig);
export const db = getFirestore(app);
export const auth = getAuth(app);
