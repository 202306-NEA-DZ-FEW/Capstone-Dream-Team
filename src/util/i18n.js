import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
