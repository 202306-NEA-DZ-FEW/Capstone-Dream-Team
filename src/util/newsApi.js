export async function fetchData(apiRoute) {
    const url = `https://newsapi.org/v2/${apiRoute}&apiKey=${process.env.NEXT_PUBLIC_NEWS_API}`;
    const response = await fetch(url);
    const data = await response.json();
    return data;
}
