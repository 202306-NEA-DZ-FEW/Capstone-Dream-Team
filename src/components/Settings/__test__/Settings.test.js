import renderer from "react-test-renderer";
import Settings from "../Settings";
import { getAuth, auth } from "@/util/firebase";

// Mock the Firebase services
jest.mock("@/util/firebase", () => ({
    getAuth: jest.fn(),
    auth: { currentUser: jest.fn() },
}));

jest.mock("next-i18next", () => ({
    useTranslation: jest.fn().mockReturnValue({ t: jest.fn() }),
}));

// Mock the useRouter hook
jest.mock("next/router", () => ({
    useRouter: jest.fn().mockReturnValue({
        push: jest.fn(),
    }),
}));

it("renders correctly", () => {
    getAuth.mockReturnValue();

    const tree = renderer.create(<Settings />).toJSON();
    expect(tree).toMatchSnapshot();
});
