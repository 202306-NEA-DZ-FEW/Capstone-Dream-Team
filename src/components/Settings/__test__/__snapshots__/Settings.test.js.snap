// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "bottom": 16,
        "left": 16,
        "pointerEvents": "none",
        "position": "fixed",
        "right": 16,
        "top": 16,
        "zIndex": 9999,
      }
    }
  />,
  <section
    className=" py-1"
  >
    <div
      className="w-full lg:w-8/12 px-4 mx-auto "
    />
    <div
      className="relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg rounded-lg  border-0"
    >
      <div
        className="rounded-t bg-white mb-0 px-6 py-6"
      >
        <div
          className="text-center flex justify-between"
        >
          <h6
            className="text-blueGray-700  tracking-wider font-light font-roboto"
          />
          <div>
            <button
              className="bg-orange-400 text-white active:bg-teal-600  uppercase text-sm px-4 py-2 rounded shadow hover:bg-orange-600 outline-none focus:outline-none mr-1 ease-linear transition-all duration-150"
              onClick={[Function]}
              type="button"
            />
          </div>
        </div>
      </div>
      <div
        className="flex-auto px-4 lg:px-10 py-10 pt-0"
      >
        <form>
          <h6
            className="text-blue-600 text-base mt-3 mb-6  "
          />
          <div
            className="flex flex-wrap sm:flex sm:flex-nowrap justify-center sm:justify-start"
          >
            <div
              className=" relative h-60 w-70 p-6 ml-4 mr-4 mb-4 border "
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <input
                className="absolute inset-0 h-60 w-70 opacity-0 z-20 "
                onChange={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                type="file"
              />
              <div
                className="text-center null"
              >
                <img
                  alt=""
                  className="mx-auto h-12 w-12"
                  src="https://www.svgrepo.com/show/357902/image-upload.svg"
                />
                <h3
                  className="mt-2 text-sm font-medium text-gray-900"
                >
                  <label
                    className="relative cursor-pointer"
                    htmlFor="file-upload"
                  >
                    <span />
                    <span
                      className="text-indigo-600"
                    >
                       
                    </span>
                    <span>
                       
                    </span>
                  </label>
                </h3>
              </div>
            </div>
            <div
              className="flex flex-wrap w-full"
            >
              <div
                className="w-full lg:w-6/12 px-4"
              >
                <div
                  className="relative w-full mb-3"
                >
                  <label
                    className="block  text-blueGray-600 text-sm  mb-2"
                    htmlfor="grid-password"
                  />
                  <input
                    className="border  px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                    name="restaurantName"
                    onChange={[Function]}
                    readOnly={true}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="w-full lg:w-6/12 px-4"
              >
                <div
                  className="relative w-full mb-3"
                >
                  <label
                    className="block  text-blueGray-600 text-sm  mb-2"
                    htmlfor="grid-password"
                  >
                     
                  </label>
                  <input
                    className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                    name="email"
                    onChange={[Function]}
                    readOnly={true}
                    type="email"
                  />
                </div>
              </div>
              <div
                className="w-full lg:w-6/12 px-4"
              >
                <div
                  className="relative w-full mb-3"
                >
                  <label
                    className="block text-blueGray-600 text-sm  mb-2"
                    htmlfor="grid-password"
                  />
                  <input
                    className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                    name="firstName"
                    onChange={[Function]}
                    readOnly={true}
                    type="text"
                    value=" "
                  />
                </div>
              </div>
              <div
                className="w-full lg:w-6/12 px-4"
              >
                <div
                  className="relative w-full mb-3"
                >
                  <label
                    className="block text-blueGray-600 text-sm  mb-2"
                    htmlfor="grid-password"
                  />
                  <input
                    className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                    name="lastName"
                    onChange={[Function]}
                    readOnly={true}
                    type="text"
                    value=" "
                  />
                </div>
              </div>
            </div>
          </div>
          <hr
            className="mt-6 border-b-1 border-blueGray-300"
          />
          <h6
            className="text-blue-600 text-base mt-3 mb-6  "
          />
          <div
            className="flex flex-wrap"
          >
            <div
              className="w-full lg:w-12/12 px-4"
            >
              <div
                className="relative w-full mb-3"
              >
                <label
                  className="block text-blueGray-600 text-sm  mb-2"
                  htmlfor="grid-password"
                />
                <input
                  className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                  name="address"
                  onChange={[Function]}
                  readOnly={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="w-full lg:w-4/12 px-4"
            >
              <div
                className="relative w-full mb-3"
              >
                <label
                  className="block text-blueGray-600 text-sm  mb-2"
                  htmlfor="grid-password"
                />
                <input
                  className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                  name="city"
                  onChange={[Function]}
                  readOnly={true}
                  type="email"
                  value=""
                />
              </div>
            </div>
            <div
              className="w-full lg:w-4/12 px-4"
            >
              <div
                className="relative w-full mb-3"
              >
                <label
                  className="block text-blueGray-600 text-sm  mb-2"
                  htmlfor="grid-password"
                />
                <input
                  className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                  name="country"
                  onChange={[Function]}
                  readOnly={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="w-full lg:w-4/12 px-4"
            >
              <div
                className="relative w-full mb-3"
              >
                <label
                  className="block text-blueGray-600 text-sm  mb-2"
                  htmlfor="grid-password"
                />
                <input
                  className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                  name="postalCode"
                  onChange={[Function]}
                  readOnly={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="w-full lg:w-4/12 px-4"
            >
              <div
                className="relative w-full mb-3"
              >
                <label
                  className="block text-blueGray-600 text-sm  mb-2"
                  htmlfor="grid-password"
                />
                <input
                  className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                  name="phoneNumber"
                  onChange={[Function]}
                  readOnly={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <hr
            className="mt-6 border-b-1 border-blueGray-300"
          />
          <h6
            className="text-blue-600 text-base mt-3 mb-6 "
          />
          <div
            className="flex flex-wrap"
          >
            <div
              className="w-full lg:w-12/12 px-4"
            >
              <div
                className="relative w-full mb-3"
              >
                <label
                  className="block text-blueGray-600 text-sm  mb-2"
                  htmlfor="grid-password"
                />
                <textarea
                  className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                  name="about"
                  onChange={[Function]}
                  readOnly={true}
                  rows="4"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </form>
        <hr
          className="mt-6 border-b-1 border-blueGray-300"
        />
        <form
          className={false}
        >
          <h6
            className="text-blue-600 text-base mt-3 mb-6  "
          />
          <div
            className="flex flex-wrap"
          >
            <div
              className="w-full lg:w-4/12 px-4"
            >
              <div
                className="relative w-full mb-3"
              >
                <label
                  className="block text-blueGray-600 text-sm  mb-2"
                  htmlfor="grid-password"
                />
                <input
                  className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                  name="password"
                  onChange={[Function]}
                  readOnly={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              className="w-full lg:w-4/12 px-4"
            >
              <div
                className="relative w-full mb-3"
              >
                <label
                  className="block text-blueGray-600 text-sm  mb-2"
                  htmlfor="grid-password"
                />
                <input
                  className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                  name="newPassword"
                  onChange={[Function]}
                  readOnly={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
            <button
              className="bg-orange-400 text-white   text-sm px-4 py-2 rounded shadow hover:bg-orange-600 outline-none focus:outline-none mr-1 ease-linear transition-all duration-150 mt-6 mb-3"
              onClick={[Function]}
              type="submit"
            />
          </div>
        </form>
        <hr
          className="mt-6 border-b-1 border-blueGray-300"
        />
        <form>
          <h6
            className="text-blue-600 text-base mt-3 mb-6  "
          />
          <div
            className="flex flex-wrap"
          >
            <div
              className="w-full lg:w-4/12 px-4"
            >
              <div
                className="relative w-full mb-3"
              >
                <label
                  className="block text-blueGray-600 text-sm  mb-2"
                  htmlfor="grid-password"
                />
                <input
                  className="border px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                  name="password"
                  onChange={[Function]}
                  readOnly={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
            <button
              className="bg-orange-400 text-white   text-sm px-4 py-2 rounded shadow hover:bg-orange-600 outline-none focus:outline-none mr-1 ease-linear transition-all duration-150 mt-6 mb-3"
              onClick={[Function]}
              type="submit"
            />
          </div>
        </form>
      </div>
    </div>
  </section>,
]
`;
